magic_admin/django/auth/backends.py

```diff
@@ -0,0 +1,146 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.backends import ModelBackend
+
+from magic_admin.django.config import MagicAuthBackendMode
+from magic_admin.django.exceptions import (
+    MissingAuthorizationHeader,
+    MissingUserEmailInput,
+    PublicAddressDoesNotExist,
+    UnsupportedAuthMode,
+    UserEmailMissmatch,
+)
+from magic_admin.error import (
+    DIDTokenError,
+    APIConnectionError,
+    RateLimitingError,
+    BadRequestError,
+    AuthenticationError,
+    ForbiddenError,
+    APIError,
+)
+
+from magic_admin.utils.http import get_identity_token_from_header
+from magic_admin.magic import Magic
+from magic_admin.utils.logging import (
+    log_debug,
+    log_info,
+)
+
+
+user_model = get_user_model()
+
+
+class MagicAuthBackend(ModelBackend):
+
+    @staticmethod
+    def _load_user_from_email(email):
+        log_debug('Loading user by email.', email=email)
+        try:
+            return user_model.objects.get(email=email)
+        except user_model.DoesNotExist:
+            return None
+
+    @staticmethod
+    def _validate_identity_token_and_load_user(
+        identity_token,
+        email,
+        public_address,
+    ):
+        try:
+            Magic().Token.validate(identity_token)
+        except (
+                DIDTokenError,
```

Comment:
Not sure about these exceptions, I just imported all the existing errors and plugged them here. I'm not sure what's the intended behavior here.